content/toolsAndResources/nge.md

```diff
@@ -0,0 +1,130 @@
+The second method is to press the spacebar for a shortcut to open a search window with a the full node list at the mouse position on the graph. Once the window is open, we can simply start typing to enter a keyword in the search bar. Once the list is filtered, use either the mouse to click the desired node or use the arrow keys to highlight the desired node and press enter. This will drop the node at the mouse position when the shortcut was invoked. If a wire was selected when the shortcut was opened, the editor will attempt to connect the node in the middle of the connection that was selected. If the node to be placed contains compatible inputs and outputs, a connection will be made.
+
+![Press spacebar for shortcut to search bar to filter node list which appears at the mouse position](/img/tools/nge/nodeAddShortcut.jpg)
+
+There is one other shortcut to quickly add nodes in a contextual manner. For most node inputs with a defined type, we can simply click on the input and drag to the left of the node which will drag out a dashed wire of the type the input expects. Once the mouse button is released, an input node of the correct type will be dropped. For example, the image below illustrates clicking on the height input of the `Box` source node and dragging a dashed line to the left of the node. Once the mouse button is released, a `Float` node is added to the graph at the mouse position.
+
+![Click and drag from node input to drop an automatically connected node of the appropriate type on release of the mouse button](/img/tools/nge/dragAddNode.jpg)
+
+## Connecting Nodes Together
+All nodes have one or more inputs and one or more outputs, however each individual node has requirements for the data that passes through it. These requirements typically take the form of data types that the nodes expect for their calculation. To easily tell what type of data a node expects is to look at the inputs and output icons on the node. To help decipher the icons, please refer to the legend below:
+
+![Input type iconography](/img/tools/nge/inputTypes.png)
+
+The first five types are self-explanatory and are the most common data types that will be seen on nodes. The last two - Geometry and Subnode - are special data types that are narrowly defined. The Geometry type is used exclusively in Node Geometry Editor and refers to the actual mesh that the node affects. The Geometry type is also the only available input for the `GeometryOutput` node which is the output for the graph which makes a geometry source - one of our simple shapes or an imported mesh - a requirement for the graph. 
+
+Subnode is rarely to found on nodes and it refers to nodes that have exactly one other node they can be connected to. Mostly this is used to reduce complexity on a large node moving optional operations or parameters to another node. While they are not commonly found on nodes, seeing the subnode icon is a good indication to look for another node that can take this node as an input.

```

Comment:
"to be found" or just "found" instead of "to found"? Sorry if what you wrote is right, it's just I never encountered such construct!

Reply:
I'm not surprised that you've not encountered the construct because it's just wrong. When I do too much rewriting and reconstructing sentences on docs this size, I tend to produce weird errors like this one and can miss them in edit because I've written it too many times and my eyes skip past them. Fixed

Reply:
updated in https://github.com/BabylonJS/Documentation/pull/963/commits/07fc54158e3862125b1824623a58ae435c2dd890